### PR TITLE
feat: add im-markdown output for doc fetch

### PR DESCRIPTION
**Pull request by liujiashu-shiro**
## Summary

Add an `im-markdown` output format for doc fetch, converting Docx content into Markdown suitable for IM messages. The change expands conversion coverage for common document structures and documents the intended `lark-doc` to `lark-im`
usage path.

## Changes

- Add Docx-to-IM-Markdown conversion logic
- Support `--doc-format im-markdown` in doc fetch
- Cover headings, lists, code blocks, tables, images, links, nested structures, and edge cases in unit tests
- Extend `docs_fetch_v2` tests for the new format behavior
- Document `im-markdown` in `lark-doc` fetch references as a fetch-only format for `lark-im` usage
- Document the `lark-im` sending workflow for forwarding fetched doc content with `--markdown`

## Test Plan

- [x] Unit tests passed: `go test ./shortcuts/doc`
- [x] Format check passed: `gofmt -l shortcuts/doc/docs_fetch_im_markdown.go shortcuts/doc/docs_fetch_im_markdown_test.go shortcuts/doc/docs_fetch_v2.go shortcuts/doc/docs_fetch_v2_test.go`
- [x] Diff whitespace check passed: `git diff --check`
- [ ] Manually verify `lark-cli docs +fetch --doc-format im-markdown` output can be sent through `lark-im` with `--markdown`

## Related Issues

None

## Summary by CodeRabbit

* **New Features**
  * Added `im-markdown` as an allowed `--doc-format` for v2 `+fetch`. It fetches as standard Markdown from the API, then converts IM-style markup (headings, callouts, blockquotes, lists, grids/columns, tables, sheets/bookmarks, and citations) into clean Markdown, including nested and partially malformed fragments.

* **Bug Fixes**
  * Improved post-processing robustness for unclosed containers and scanner/attribute edge cases, preserving or safely dropping fragments as appropriate.

* **Tests**
  * Expanded unit and integration-style coverage for request construction/downgrades and Markdown conversion behaviors, including escaping.

* **Documentation**
  * Updated `lark-doc`/`lark-im` docs to clarify `im-markdown` is fetch-only and how to send converted content as a message.

### CODE REVIEW

File: skills/lark-doc/SKILL.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ lark-cli docs +update --api-version v2 --doc "文档URL或token" --command appen
 
 **CRITICAL — 执行对应操作前，MUST 先用 Read 工具读取以下文件，缺一不可：**
 1. [`../lark-shared/SKILL.md`](../lark-shared/SKILL.md) — 认证、权限处理、全局参数（所有操作通用）
-2. **读取文档（`docs +fetch --api-version v2`）** → 必读 [`lark-doc-fetch.md`](references/lark-doc-fetch.md)（`--scope` / `--detail` 选择、局部读取策略、`<fragment>` / `<excerpt>` 输出结构）
+2. **读取文档（`docs +fetch --api-version v2`）** → 必读 [`lark-doc-fetch.md`](references/lark-doc-fetch.md)（`--scope` / `--detail` / `--doc-format` 选择、局部读取策略、`<fragment>` / `<excerpt>` 输出结构）
```
Comment (SunPeiYang996) [inline skills/lark-doc/SKILL.md]: 不用改

File: skills/lark-doc/SKILL.md [hunk 2]
```diff
@@ -67,7 +67,7 @@ Shortcut 是对常用操作的高级封装（`lark-cli docs +<verb> [flags]`）
 | Shortcut | 说明 |
 |----------|------|
 | [`+create`](references/lark-doc-create.md) | Create a Lark document (XML / Markdown) |
-| [`+fetch`](references/lark-doc-fetch.md) | Fetch Lark document content (XML / Markdown) |
+| [`+fetch`](references/lark-doc-fetch.md) | Fetch Lark document content (XML / Markdown / text; `im-markdown` only after fetch for `lark-im`) |
```
Comment (SunPeiYang996) [inline skills/lark-doc/SKILL.md]: text 去掉吧，笔误，后面我统一去掉 skill 里的 text

**Review by SunPeiYang996 — APPROVED** [review]
